generate-component/index.js

```diff
@@ -0,0 +1,77 @@
+const fs = require("fs");
+const { component, story, test, barrel } = require("./component_templates.tsx");
+
+// grab component name from terminal argument
+const [name] = process.argv.slice(2);
+if (!name) throw new Error("You must include a component name.");
+
+const uppercasedName =
+  name.charAt(0).toUpperCase() + name.slice(1).toLowerCase();
+
+const dir = `./packages/syntax-core/src/${uppercasedName}/`;
+
+// throw an error if the file already exists
+if (fs.existsSync(dir))
```

Comment:
@somethiiing can we convert the `fs.*` to the async / promise versions?

Reply:
will look into this